cookbook/bundles/tests.md

```diff
@@ -0,0 +1,255 @@
+
+    require_once __DIR__.'/autoload.php';
+
+    $kernel = new AppKernel('prod', false);
+    $request = Request::createFromGlobals();
+    $response = $kernel->handle($request);
+    $response->send();
+
+With this it becomes possible to browse the page. A way to do it without having to configure
+a web server is to run the following command:
+
+    php Tests/app/console.php server:run -d Tests/app
+
+.. note::
+
+    If the third party bundle uses the Twig templating engine to render HTML pages
+    or if it uses the Form Component or anything else, then more dependencies and
+    configuration parameters should be added.
+
+Automated tests
+---------------
+
+Manual tests are great to get a quick idea of what the bundle does. But writing
+automated tests is even better!
+
+The first step is to install a test framework like PHPUnit:
+
+    composer require --dev phpunit/phpunit
+
+.. note::
+
+    The steps should be similar with other tests frameworks.
+
+Then the second one is to configure it to use composer's autoloading:
+
+    <?xml version="1.0" encoding="UTF-8"?>
+
+    <!-- http://phpunit.de/manual/4.3/en/appendixes.configuration.html -->
+    <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.3/phpunit.xsd"
+        backupGlobals="false"
+        colors="true"
+        bootstrap="./vendor/autoload.php"
+    >
+        <testsuites>
+            <testsuite name="Test Suite">
+                <directory>./Tests/</directory>
+            </testsuite>
+        </testsuites>
+    </phpunit>
+
+With these two simple steps it becomes possible to run the test suite with the following command:
+
+    vendor/bin/phpunit
+
+Functional CLI tests
+~~~~~~~~~~~~~~~~~~~~
+
+As advised in the official best practices (smoke testing), writing tests for
+commands can be done by simply checking the exit code:
+
+    <?php
+    // File: Tests/Command/DemoCommandTest.php
+
+    namespace Acme\MyBundle\Tests\Command;
+
+    use Acme\MyBundle\Tests\AppKernel;
+    use Symfony\Bundle\FrameworkBundle\Console\Application;
+    use Symfony\Component\Console\Tester\ApplicationTester;
+
+    class DemoCommandTest extends \PHPUnit_Framework_TestCase
+    {
+        private $application;
+
+        protected function setUp()
+        {
+            $kernel = new AppKernel('test', false);
+            $application = new Application($kernel);
+            $application->setAutoExit(false);
+            $this->application = new ApplicationTester($application);
+        }
+
+        public function testItRunsSuccessfully()
+        {
+            $exitCode = $this->application->run(array(
+                'command_name' => 'acme:demo',
+                'argument' => 'value',
+                '--option' => 'value',
+            ));
+
+            $this->assertSame(0, $exitCode, $this->application->getDisplay());
+        }
+    }
+
+Functional web tests
```

Comment:
as controllers are more often tested than commands, I would put it before the commands section.